### DISCUSSION TITLE
Latest.data is'nt loading from cache

### DISCUSSION THREAD

**Discussion by avneesh70** [jeffreylanters/react-unity-webgl#546] [Help Wanted!]
I'm trying to load assets from my node server wasm gets cached as expected but data gets downloaded every time page is reloaded. I'm using it this way:

```
  function handleCacheControl(url) {
    if (url.match(/\.data/) || url.match(/\.bundle/) || url.match(/\.wasm/)) {
      return "must-revalidate";
    }
    if (url.match(/\.mp4/) || url.match(/\.wav/)) {
      return "immutable";
    }
    return "no-store";
  }
  const { unityProvider, isLoaded, loadingProgression, sendMessage } =
    useUnityContext({
      loaderUrl: `${import.meta.env.VITE_API_SPEECH}/Build/Latest.loader.js`,
      dataUrl: `${import.meta.env.VITE_API_SPEECH}/Build/Latest.data`,
      frameworkUrl: `${import.meta.env.VITE_API_SPEECH}/Build/Latest.framework.js`,
      codeUrl: `${import.meta.env.VITE_API_SPEECH}/Build/Latest.wasm`,
      cacheControl: handleCacheControl,
      // streamingAssetsUrl: "streamingassets",
    });
```
and my backend public contains react build:

```
  app.use(
    express.static(path.resolve(__dirname, "public"), { extensions: ["js"] })
  );

  app.use("/", function (_, response) {
    response.sendFile(path.join(__dirname, "public", "index.html"));
  });
```

**Comment by samijoehayek** [comment]
Have you been able to resolve the issue? I am facing the same problem!

**Reply by avneesh70** [reply]
not yet i did a roundabout to use gzip to compress that file although there's not much difference on .data one.

**Reply by samijoehayek** [reply]
I have opened a ticket strictly explaining the problem, the thing is if you run the index file on its own on the server, the same one serving the frontend app, it will cache normally but when its just when u put the build file in a nextjs or reactjs app it will not cache i have tested it.

What kind of js library are you using? react, next?

**Reply by avneesh70** [reply]
ic.. im using react for frontend and express on backend

EDIT: 

> if you run the index file on its own on the server, the same one serving the frontend app, it will cache normally

what if i serve react build from express would it work?

**Reply by avneesh70** [reply]
@samijoehayek I got the fix, just uncheck Data Caching in Unity Build settings->Publishing settings

![image](https://github.com/jeffreylanters/react-unity-webgl/assets/108516456/1fe8932b-3a04-4986-9b14-848bd02a9cc0)

**Reply by samijoehayek** [reply]
Ill Check it out right now and let you know.
